### PR TITLE
UCT/UD: fixes problem with the allocation of reserved skbs

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe

### CODE REVIEW

**Comment by yosefe** [comment]
@alex-mikheev pls add unit test to reproduce the issue

**Comment by shamisp** [comment]
why not x4 ? please add documentation

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/891/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1502/ for details.

File: src/uct/ib/ud/base/ud_iface.c [hunk 1]
```diff
@@ -305,6 +305,14 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, uct_ud_iface_config_t *config)
     return UCS_ERR_INVALID_PARAM;
 }
 
+void uct_ud_iface_complete_init(uct_worker_h worker, uct_ud_iface_t *iface,
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.c]: can we move the timer creation here as well so ud constructores would not have to take a lock?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1520/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/910/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/913/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1522/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/923/ for details (Mellanox internal link).

File: src/uct/ib/ud/base/ud_iface.c [hunk 2]
```diff
@@ -305,6 +305,27 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, uct_ud_iface_config_t *config)
     return UCS_ERR_INVALID_PARAM;
 }
 
+void uct_ud_iface_complete_init(uct_ud_iface_t *iface, ucs_notifier_chain_func_t progress_cb)
+{
+    ucs_status_t status;
+
+    /* TODO: add progress on first ep creation */
+    uct_ud_enter(iface);
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.c]: @alex-mikheev adding here the what we talked f2f- we could avoid taking the lock here if adding the timer would be the very last operation done in the ud iface initialization

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1553/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/946/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1:

**Pull request merged**